### ISSUE TITLE
Training Speed Varying

### ISSUE THREAD

**Issue by melights** [mrharicot/monodepth#111]
Hi Mrharicot,

Thank you for sharing this great work. When I am training the monodepth model on KITTI dataset using your script on a single GPU, I am experiencing training speed varying problem. Do you have any idea why this happens? I have beeing debugging these for several days but no luck. The dataloader pipeline seems to be already asynchronous.

batch 182000 | examples/s: 5.29 | loss: 1.22465 | time elapsed: 62.35h | time left: 61.84h
batch 182100 | examples/s: 1.41 | loss: 1.43983 | time elapsed: 62.41h | time left: 61.83h
batch 182200 | examples/s: 1.83 | loss: 1.06617 | time elapsed: 62.47h | time left: 61.82h
batch 182300 | examples/s: 1.22 | loss: 1.91274 | time elapsed: 62.53h | time left: 61.81h
batch 182400 | examples/s: 1.25 | loss: 1.44637 | time elapsed: 62.58h | time left: 61.79h
batch 182500 | examples/s: 1.47 | loss: 1.23239 | time elapsed: 62.64h | time left: 61.78h
batch 182600 | examples/s: 1.27 | loss: 1.18121 | time elapsed: 62.70h | time left: 61.77h
batch 182700 | examples/s: 3.01 | loss: 1.22865 | time elapsed: 62.74h | time left: 61.75h
batch 182800 | examples/s: 0.95 | loss: 1.12455 | time elapsed: 62.80h | time left: 61.74h
batch 182900 | examples/s: 1.76 | loss: 1.59430 | time elapsed: 62.86h | time left: 61.72h
batch 183000 | examples/s: 1.24 | loss: 1.47687 | time elapsed: 62.92h | time left: 61.71h
batch 183100 | examples/s: 6.38 | loss: 1.58245 | time elapsed: 62.97h | time left: 61.70h
batch 183200 | examples/s: 1.08 | loss: 1.16879 | time elapsed: 63.02h | time left: 61.68h
batch 183300 | examples/s: 1.40 | loss: 1.03136 | time elapsed: 63.06h | time left: 61.65h
batch 183400 | examples/s: 6.52 | loss: 1.52166 | time elapsed: 63.12h | time left: 61.64h
batch 183500 | examples/s: 1.85 | loss: 1.26706 | time elapsed: 63.17h | time left: 61.62h
batch 183600 | examples/s: 6.11 | loss: 1.25645 | time elapsed: 63.20h | time left: 61.59h
batch 183700 | examples/s: 2.87 | loss: 1.42925 | time elapsed: 63.27h | time left: 61.58h
batch 183800 | examples/s: 1.40 | loss: 1.32448 | time elapsed: 63.32h | time left: 61.57h
batch 183900 | examples/s: 5.79 | loss: 1.51024 | time elapsed: 63.38h | time left: 61.55h
batch 184000 | examples/s: 1.36 | loss: 1.36937 | time elapsed: 63.44h | time left: 61.54h
batch 184100 | examples/s: 1.82 | loss: 1.12686 | time elapsed: 63.49h | time left: 61.53h
batch 184200 | examples/s: 3.96 | loss: 1.43068 | time elapsed: 63.55h | time left: 61.52h
batch 184300 | examples/s: 1.73 | loss: 2.18362 | time elapsed: 63.58h | time left: 61.47h
batch 184400 | examples/s: 2.14 | loss: 1.74562 | time elapsed: 63.64h | time left: 61.47h
batch 184500 | examples/s: 1.25 | loss: 1.16859 | time elapsed: 63.70h | time left: 61.46h
batch 184600 | examples/s: 2.52 | loss: 1.45383 | time elapsed: 63.76h | time left: 61.44h
batch 184700 | examples/s: 2.68 | loss: 1.59525 | time elapsed: 63.81h | time left: 61.42h
batch 184800 | examples/s: 1.31 | loss: 1.41034 | time elapsed: 63.86h | time left: 61.41h
batch 184900 | examples/s: 1.87 | loss: 1.74731 | time elapsed: 63.91h | time left: 61.38h
batch 185000 | examples/s: 1.58 | loss: 1.20761 | time elapsed: 64.00h | time left: 61.41h
batch 185100 | examples/s: 3.06 | loss: 1.59985 | time elapsed: 64.08h | time left: 61.41h
batch 185200 | examples/s: 2.22 | loss: 1.61685 | time elapsed: 64.12h | time left: 61.39h
batch 185300 | examples/s: 1.84 | loss: 0.90981 | time elapsed: 64.16h | time left: 61.35h
batch 185400 | examples/s: 2.58 | loss: 1.22769 | time elapsed: 64.21h | time left: 61.34h
batch 185500 | examples/s: 2.59 | loss: 1.90655 | time elapsed: 64.27h | time left: 61.32h
batch 185600 | examples/s: 2.09 | loss: 1.05945 | time elapsed: 64.32h | time left: 61.31h
batch 185700 | examples/s: 6.22 | loss: 1.06977 | time elapsed: 64.36h | time left: 61.28h
batch 185800 | examples/s: 1.21 | loss: 1.17538 | time elapsed: 64.41h | time left: 61.26h
batch 185900 | examples/s: 0.90 | loss: 1.25430 | time elapsed: 64.50h | time left: 61.28h
batch 186000 | examples/s: 6.55 | loss: 1.42911 | time elapsed: 64.53h | time left: 61.24h
batch 186100 | examples/s: 6.27 | loss: 1.34087 | time elapsed: 64.55h | time left: 61.19h
batch 186200 | examples/s: 6.58 | loss: 1.06404 | time elapsed: 64.57h | time left: 61.14h
batch 186300 | examples/s: 6.36 | loss: 1.66774 | time elapsed: 64.59h | time left: 61.09h
batch 186400 | examples/s: 6.53 | loss: 1.22544 | time elapsed: 64.61h | time left: 61.04h
batch 186500 | examples/s: 6.42 | loss: 1.89006 | time elapsed: 64.63h | time left: 60.99h
batch 186600 | examples/s: 6.10 | loss: 1.42371 | time elapsed: 64.64h | time left: 60.94h
batch 186700 | examples/s: 6.70 | loss: 1.30297 | time elapsed: 64.66h | time left: 60.89h
batch 186800 | examples/s: 6.04 | loss: 1.64698 | time elapsed: 64.68h | time left: 60.84h
batch 186900 | examples/s: 6.35 | loss: 1.14622 | time elapsed: 64.70h | time left: 60.79h
batch 187000 | examples/s: 6.01 | loss: 0.94734 | time elapsed: 64.71h | time left: 60.74h
batch 187100 | examples/s: 6.11 | loss: 1.53086 | time elapsed: 64.73h | time left: 60.68h
batch 187200 | examples/s: 6.23 | loss: 1.84523 | time elapsed: 64.75h | time left: 60.63h

**Comment by mrharicot** [comment]
Hi,
What is your GPU?
What is your dataset?
Is it stored on a SSD?

**Comment by melights** [comment]
Hi mrharicot,

It is old version of Titan X.
Dataset is KITTI.
Not stored on a SSD.

I changed the loss function a bit, so 6 examples/s should be a normal speed. But it constantly jumps to 1 or 2 examples/s.

**Comment by mrharicot** [comment]
Hi,
You should store your data on an SSD, this is probably the cause of the slow speed.
HDDs are quite bad at loading lots of small files from random places on the disk.
